### PR TITLE
Make kubediff easier to integrate with deploy.

### PR DESCRIPTION
**Pull request by tomwilkie**
This PR makes it easier to integration with other systems (deploy).

- When using JSON output, don't use the exit value to indicate presence of diffs, so we can treat the exit code as just errors, and parse the output properly.
- Key JSON output by filename
- Make JSON output include the correct error strings
- Don't fail when file doesn't exist

### CODE REVIEW

**Comment by ekimekim** [comment]
nit: Instead of coupling this to the output format, I'd rather see it as a seperate flag eg. `--error-status-only`.
That said, happy to merge as is. Up to you.

**Comment by tomwilkie** [comment]
Yeah thats a better idea.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Always prefer setuptools over distutils
+from setuptools import setup, find_packages
+# To use a consistent encoding
+from codecs import open
+from os import path
+
+here = path.abspath(path.dirname(__file__))
```
Comment (ekimekim) [inline setup.py]: i'm like 99% sure that defining `here` and importing a special `open` doesn't do anything. Most likely whatever template or tutorial you were looking at were expecting you to do something like `long_description=open(path.join(here, 'README'))`.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Always prefer setuptools over distutils
+from setuptools import setup, find_packages
```
Comment (ekimekim) [inline setup.py]: you can use `packages=find_packages()` below (my preference, it's one less repetition of data that can go wrong), but if you aren't can you please cut this unused import?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    # Author details
+    author='Weaveworks',
+    author_email='help@weave.works',
+
+    # Choose your license
+    license='MIT',
+
+    # You can just specify the packages manually here if your project is
+    # simple. Or you can use find_packages().
+    packages=["kubedifflib"],
```
Comment (ekimekim) [inline setup.py]: You should add `install_requires=['PyYAML']` to your setup args here, as it's a hard dependency of the library.

**Pull request merged**